lib/endpoints/class-wp-json-controller.php

```diff
@@ -103,4 +103,21 @@ public function get_item_schema() {
 		return array();
 	}
 
+	/**
+	 * Get an array of endpoint arguments from the item schema for the controller.
+	 * 
+	 * @return array
+	 */
+	public function get_endpoint_args_for_item_schema() {
```

Comment:
Currently posts is the only one that is using args from the schema so far I believe, but this is in the base controller so all others can user it (so that's at least 3 times per object type). Also, from my commit message: This Pull Request changes that to be `array( 'title' => array( ... ), ... )` which can / will be extended to add arg validation per the enums / datatype defines in the schema.